lib/honeybadger/agent.rb

```diff
@@ -404,7 +404,21 @@ def event(event_type, payload = {})
         with_error_handling { hook.call(event) }
       end
 
-      return if config.ignored_events.any? { |check| event.event_type&.match?(check) }
+      return if config.ignored_events.any? do |check|
+        if check.is_a?(String) || check.is_a?(Regexp)
+          event.event_type&.match?(check)
+        elsif check.is_a?(Hash)
+          flat_hash(check).any? do |keys, value|
+            sym_keys = keys.map(&:to_sym)
```

Comment:
This allows to query deeply nested payloads.

However, I'm not sure how I feel about this. I'm expecting all keys to be a symbol, which feels wrong since an event payload can contain any mix of string/symbol keys.

Reply:
If you don't convert to symbols, this should still work and allow string keys.

Reply:
I think the one issue is, when we configure the gem using `honeybadger.yml`, the keys are strings.

For example, if I have this in the config:
```yaml
events:
  ignore:
    - foo: 'bar'
```

And create an event like this:
```ruby
Honeybadger.event(foo: 'bar')
```

I'd need to convert the keys to match.

One thing to note, I'm going through this because I want to easily call `#dig` on the payload to get the value. Perhaps I should just do some iteration to check both string and symbol keys. Gonna think about this for a bit.